### PR TITLE
Make sched deployment checking more time robust

### PR DESCRIPTION
**Pull request by jlojosnegros**
Follow-up for https://github.com/openshift-kni/numaresources-operator/pull/100#discussion_r786541738 where it was found this check could fail due to timeout.
Wrapping scheduler deployment check in an Eventually block make it more resilience to a slow CI

### CODE REVIEW

File: test/e2e/sched/install/install_test.go [hunk 1]
```diff
@@ -75,9 +75,21 @@ var _ = Describe("[Scheduler] install", func() {
 			Expect(err).NotTo(HaveOccurred())
 
 			By("checking the NumaResourcesScheduler Deployment is correctly deployed")
-			deploy, err := schedutils.GetDeploymentByOwnerReference(nroSchedObj.UID)
-			Expect(err).NotTo(HaveOccurred())
-			Expect(deploy.Status.ReadyReplicas).To(BeIdenticalTo(int32(1)))
+			const DeploymentCheckTimeout = 5 * time.Minute
+			const DeploymentCheckPollPeriod = 10 * time.Second
```
Comment (fromanirh) [inline test/e2e/sched/install/install_test.go:79]: C++-ism :) I'd rather avoid it but not a big deal

File: test/e2e/sched/install/install_test.go [hunk 2]
```diff
@@ -75,9 +75,21 @@ var _ = Describe("[Scheduler] install", func() {
...
+			Eventually(func() bool {
+				deploy, err := schedutils.GetDeploymentByOwnerReference(nroSchedObj.UID)
+				if err != nil {
+					klog.Warningf("unable to get deployment by owner reference: %v", err)
+					return false
+				}
+
+				if deploy.Status.ReadyReplicas != 1 {
```
Comment (fromanirh) [inline test/e2e/sched/install/install_test.go:87]: Let's not hardcodec constants: let's check the ready replicas are equal to the expected replicas (can't remember the field name atm)

File: test/e2e/sched/install/install_test.go (hunk 1 above)
Comment (fromanirh) [reply test/e2e/sched/install/install_test.go:79]: on second thought we can also keep them
not a big deal

**Review by fromanirh — COMMENTED** [review]
it's almost ready, let's just not hardcoded the expected replica count. Everything else is actually ok as it is.

File: test/e2e/sched/install/install_test.go (hunk 1 above)
Comment (jlojosnegros) [reply test/e2e/sched/install/install_test.go:79]: maybe I should change the names to start with a lowercase letter to be more go-ish style

Comment (fromanirh) [reply test/e2e/sched/install/install_test.go:79]: I think yes, but it's minor.

File: test/e2e/sched/install/install_test.go (hunk 2 above)
Comment (jlojosnegros) [reply test/e2e/sched/install/install_test.go]: ok

**Comment by fromanirh** [comment]
/lgtm
/approve
we have some utilities which do the same check, and which may want to reuse, but better consolidate with a later PR

**Comment by jlojosnegros** [comment]
> /lgtm /approve we have some utilities which do the same check, and which may want to reuse, but better consolidate with a later PR

Sure, once #95 is merged we could reconsider all these tests, and maybe consolidate all the different "utilities" folders under test.

**Pull request merged**